GiniVision/Classes/Networking/GiniNetworkingScreenAPICoordinator.swift

```diff
@@ -136,23 +135,44 @@ final class GiniNetworkingScreenAPICoordinator: GiniScreenAPICoordinator {
 // MARK: - Networking methods
 
 extension GiniNetworkingScreenAPICoordinator {
+    
+    @objc func showDigitalInvoiceScreen(extractionResult: ExtractionResult, analysisDelegate: AnalysisDelegate) throws {
+    
+        let digitalInvoiceViewController = DigitalInvoiceViewController()
+        digitalInvoiceViewController.giniConfiguration = giniConfiguration
+        digitalInvoiceViewController.invoice = try DigitalInvoice(extractionResult: extractionResult)
+        digitalInvoiceViewController.delegate = self
+        digitalInvoiceViewController.analysisDelegate = analysisDelegate
+        
+        screenAPINavigationController.pushViewController(digitalInvoiceViewController, animated: true)
+    }
+    
     fileprivate func startAnalysis(networkDelegate: GiniVisionNetworkDelegate) {
         self.documentService.startAnalysis { result in
+            
             switch result {
-            case .success(let extractions):
-                self.deliver(result: extractions, analysisDelegate: networkDelegate)
+            case .success(let extractionResult):
+                
+                DispatchQueue.main.async {
+                    do {
+                        try self.showDigitalInvoiceScreen(extractionResult: extractionResult, analysisDelegate: networkDelegate)
+                    } catch {
+                        self.deliver(result: extractionResult, analysisDelegate: networkDelegate)
+                    }
```

Comment:
Could you make the decision to show the digital invoice screen more explicit? I did it by checking whether the `lineItems` compound extraction label is present or not.

You can still keep the `do - catch` of course.

Reply:
Updated. Does showing DigitalInvoice initialisation show better what's going on?

Reply:
Thank you! Looks great!